lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4508,6 +4934,7 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
     TC.checkDeclAttributes(SD);
     checkAccessControl(TC, SD);
+    checkUsableFromInline(TC, SD);
```

Comment:
Do we need to do both? Can't `checkAccessControl` handle it?

Reply:
Yes, I could merge the two into one function. It would eliminate the duplicated logic for visting the TypeLocs, at the cost of making the whole function longer. However, we could then split it up into smaller functions for each Decl kind or something, maybe even turn it into a visitor. Sounds good?